woodwork/table_accessor.py

```diff
@@ -145,6 +146,27 @@ def __getattr__(self, attr):
         else:
             raise AttributeError(f"Woodwork has no attribute '{attr}'")
 
+    def __getitem__(self, key):
+        if isinstance(key, list):
+            index = pd.Index(key)
+            columns = self._dataframe.columns
+            diff = index.difference(columns)
+            in_columns = diff.isin(columns).all()
+
+            if not in_columns:
+                raise KeyError(f"Column(s) '{list(diff)}' not found in DataTable")
+
+            return self._get_subset_df_with_schema(key)
+
+        if key not in self._dataframe:
+            raise KeyError(f"Column with name '{key}' not found in DataTable")
```

Comment:
`DataTable` -> `DataFrame`

Reply:
Can we use `ColumnNotPresentError` error?
- https://github.com/alteryx/woodwork/blob/f18395909c57061415242b8b65cabed99477bd5c/woodwork/exceptions.py#L47
- We can modify that custom Exception to have an argument, something like this:

```python
class ColumnNotPresentError(KeyError):
    def __init__(self, column):
        message = "Column with name '{key}' not found in DataTable".format(key=column)
        super().__init__(self.message)
```
- Then raise the custom exception like this
```python
    raise ColumnNotPresentError(key)
```

Reply:
✅  c488348